### PR TITLE
[RLlib; setup] Remove unnecessary dependencies from `setup.py`; add `ormsgpack` dependency for RLlib

### PR DESCRIPTION
**Pull request by sven1977**
* Remove unnecessary RLlib dependencies from `setup.py` (rich, type, scikit-image).
* Add `ormsgpack` dependency for RLlib.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -315,11 +315,9 @@ def get_packages(self):
         "dm_tree",
         "gymnasium==1.0.0",
         "lz4",
-        "scikit-image",
+        "ormsgpack==1.7.0",
```
Comment (sven1977) [inline python/setup.py:318]: Get rid of 3, add 1 :)

Comment (simonsays1980) [reply python/setup.py:318]: @sven1977 Do you think we should use `ormsgpack` for our offline RL API? We use right now `msgpack` with `msgpack-numpy`.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Short question in reagrd to offline RL API.

File: python/setup.py (hunk 1 above)
Comment (sven1977) [reply python/setup.py:318]: I'm not 100% sure yet. This is still in experimental stage. `ormsgpack` is faster and natively supports numpy. That's good. But `msgpack-numpy` might produce smaller files. We'll have to see what outweighs what.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**